plugins/modules/homebrew_services.py

```diff
@@ -0,0 +1,279 @@
+              providing an alternative C(brew) path enables managing different set of packages in an alternative location in the system."
+        default: '/usr/local/bin:/opt/homebrew/bin:/home/linuxbrew/.linuxbrew/bin'
+        type: path
+    state:
+        description:
+            - State of the package's service.
+        choices: [ 'present', 'absent', 'restarted' ]
+        default: present
+        type: str
+"""
+
+EXAMPLES = """
+- name: Install foo package
+  community.general.homebrew:
+    name: foo
+    state: present
+
+- name: Start the foo service (equivalent to `brew services start foo`)
+  community.general.homebrew_service:
+    name: foo
+    state: present
+
+- name: Restart the foo service (equivalent to `brew services restart foo`)
+  community.general.homebrew_service:
+    name: foo
+    state: restarted
+
+- name: Remove the foo service (equivalent to `brew services stop foo`)
+  community.general.homebrew_service:
+    name: foo
+    service_state: absent
+"""
+
+RETURN = """
+msg:
+    description:
+      - Whether the service is now running, its PID if it is running,
+        and whether the running state changed (including restarts).
+    returned: always
+    type: str
+    sample: "Running: true, Changed: false, PID: 1234"
+pid:
+    description:
+      - If the service is now running, this is the PID of the service, otherwise -1.
+    returned: success
+    type: int
+    sample: 1234
+running:
+    description:
+      - Whether the service is running after running this command.
+    returned: success
+    type: bool
+    sample: true
+changed:
+    description:
+      - Whether the service state changed. This is true when an already running service is restarted.
+    returned: success
+    type: bool
+    sample: false
+"""
+
+import json
+import sys
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.homebrew import (
+    HomebrewValidate,
+    parse_brew_path,
+)
+
+if sys.version_info < (3, 5):
+    from collections import namedtuple
+
+    # Stores validated arguments for an instance of an action.
+    # See DOCUMENTATION string for argument-specific information.
+    HomebrewServiceArgs = namedtuple(
+        "HomebrewServiceArgs", ["name", "state", "brew_path"]
+    )
+
+    # Stores the state of a Homebrew service.
+    HomebrewServiceState = namedtuple("HomebrewServiceState", ["running", "pid"])
+
+else:
+    from typing import NamedTuple, Optional
+
+    # Stores validated arguments for an instance of an action.
+    # See DOCUMENTATION string for argument-specific information.
+    HomebrewServiceArgs = NamedTuple(
+        "HomebrewServiceArgs", [("name", str), ("state", str), ("brew_path", str)]
+    )
+
+    # Stores the state of a Homebrew service.
+    HomebrewServiceState = NamedTuple(
+        "HomebrewServiceState", [("running", bool), ("pid", Optional[int])]
+    )
```

Comment:
I'm not worried about having control of the types passed in. NamedTuple is purely to help IDEs with type checking. It is simply a gift for myself and others who love their statically checked python

Reply:
We should throw a party the day old Pythons are no longer supported, and in that party we should add type hints/checks throughout the modules, plugins, collections, everything. You and the others are definitely invited :-)
That being said, though, I have mixed feelings about having this conditional typing around.

Reply:
> We should throw a party the day old Pythons are no longer supported, and in that party we should add type hints/checks throughout the modules, plugins, collections, everything.

That will be the case for communiy.general 12.0.0, to be released in fall 2025. (That release will require ansible-core 2.17+, which requires Python 3.7+ on the target and Python 3.10+ on the controller.)

(Actually for everything that's not in plugins/modules/ and plugins/module_utils/ we're already at Python 3.8+, and from the next major release on at Python 3.9+. It would have already been 9.0.0 if I hadn't forgotten about that... So if someone is sufficiently bored... :D)

> That being said, though, I have mixed feelings about having this conditional typing around.

I agree. On the one hand it will make it easier to type things eventually, since you only have to delete the other branch and the `if`, but on the other hand it complicates the code currently.